### PR TITLE
Fix collection 4: substituteInPlace cleanups, pythonPackages.azure-synapse-artifacts: Fix meta

### PR DESCRIPTION
**Pull request by SuperSandro2000**
###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [ ] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/development/python-modules/azure-synapse-artifacts/default.nix [hunk 1]
```diff
@@ -23,8 +23,8 @@ buildPythonPackage rec {
   pythonImportsCheck = [ "azure.synapse.artifacts" ];
 
   meta = with lib; {
-    description = "CHANGE";
-    homepage = "https://github.com/CHANGE/azure-synapse-artifacts/";
+    description = "Microsoft Azure Synapse Artifacts Client Library for Python";
+    homepage = "https://github.com/Azure/azure-sdk-for-python";
```
Comment (jonringer) [inline pkgs/development/python-modules/azure-synapse-artifacts/default.nix:27]: oops... thanks

**Review by jonringer — APPROVED** [review]
diff LGTM

**Comment by SuperSandro2000** [comment]
This is a semi-automatic executed nixpkgs-review which is checked by a human on a best effort basis and does not build all packages (e.g. lumo, tensorflow or pytorch).
If you have any questions or problems please reach out to SuperSandro2000 on IRC.

Result of `nixpkgs-review pr 110210` run on x86_64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
<summary>
6 packages marked as broken and skipped:
</summary>
<ul>
<li>python37Packages.bellows</li>
<li>python37Packages.optuna</li>
<li>python38Packages.bellows</li>
<li>python38Packages.optuna</li>
<li>python39Packages.bellows</li>
<li>python39Packages.google-music</li>
</ul>
</details>
<details>
<summary>
23 packages built:
</summary>
<ul>
<li>appdaemon</li>
<li>gdbgui</li>
<li>google-music-scripts</li>
<li>python37Packages.audio-metadata</li>
<li>python37Packages.bidict</li>
<li>python37Packages.flask-socketio</li>
<li>python37Packages.google-music</li>
<li>python37Packages.google-music-proto</li>
<li>python37Packages.google-music-utils</li>
<li>python37Packages.python-socketio</li>
<li>python38Packages.audio-metadata</li>
<li>python38Packages.bidict</li>
<li>python38Packages.flask-socketio</li>
<li>python38Packages.google-music</li>
<li>python38Packages.google-music-proto</li>
<li>python38Packages.google-music-utils</li>
<li>python38Packages.python-socketio</li>
<li>python39Packages.audio-metadata</li>
<li>python39Packages.bidict</li>
<li>python39Packages.flask-socketio</li>
<li>python39Packages.google-music-proto</li>
<li>python39Packages.google-music-utils</li>
<li>python39Packages.python-socketio</li>
</ul>
</details>

**Comment by SuperSandro2000** [comment]
This is a semi-automatic executed nixpkgs-review which is checked by a human on a best effort basis and does not build all packages (e.g. lumo, tensorflow or pytorch).
If you have any questions or problems please reach out to SuperSandro2000 on IRC.

Result of `nixpkgs-review pr 110210` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
<summary>
2 packages marked as broken and skipped:
</summary>
<ul>
<li>python37Packages.optuna</li>
<li>python38Packages.optuna</li>
</ul>
</details>
<details>
<summary>
30 packages built:
</summary>
<ul>
<li>appdaemon</li>
<li>gdbgui</li>
<li>google-music-scripts</li>
<li>python37Packages.audio-metadata</li>
<li>python37Packages.bellows</li>
<li>python37Packages.bidict</li>
<li>python37Packages.flask-socketio</li>
<li>python37Packages.google-music</li>
<li>python37Packages.google-music-proto</li>
<li>python37Packages.google-music-utils</li>
<li>python37Packages.mxnet</li>
<li>python37Packages.python-socketio</li>
<li>python37Packages.spyder</li>
<li>python38Packages.audio-metadata</li>
<li>python38Packages.bellows</li>
<li>python38Packages.bidict</li>
<li>python38Packages.flask-socketio</li>
<li>python38Packages.google-music</li>
<li>python38Packages.google-music-proto</li>
<li>python38Packages.google-music-utils</li>
<li>python38Packages.mxnet</li>
<li>python38Packages.python-socketio</li>
<li>spyder (python38Packages.spyder)</li>
<li>python39Packages.audio-metadata</li>
<li>python39Packages.bellows</li>
<li>python39Packages.bidict</li>
<li>python39Packages.flask-socketio</li>
<li>python39Packages.google-music-proto</li>
<li>python39Packages.mxnet</li>
<li>python39Packages.python-socketio</li>
</ul>
</details>

**Pull request merged**